_posts/2019-07-08-persistent-storage-improvements.md

```diff
@@ -0,0 +1,261 @@
+We tested this feature with Kubernetes and OpenShift on Amazon AWS.
+It should be compatible with Kubernetes 1.11+ and OpenShift 3.11+.
+Resizing persistent volumes should work on most major public clouds (such as Amazon AWS, Microsoft Azure and Google Cloud) and many other storage types (such as Cinder or Ceph).
+
+To tell your Kubernetes or OpenShift cluster that your storage supports volume resizing, you have to set the `allowVolumeExpansion` option in your StorageClass to `true`.
+For example following StorageClass would create Amazon AWS GP2 volumes with `xfs` filesystem, encryption and enabled volume expansion:
+
+```yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1beta1
+metadata:
+  name: ssd
+  annotations:
+    storageclass.kubernetes.io/is-default-class: "true"
+provisioner: kubernetes.io/aws-ebs
+parameters:
+  type: gp2
+  encrypted: "true"
+  fsType: "xfs"
+reclaimPolicy: Delete
+allowVolumeExpansion: true
+```
+
+The option is called `allowVolumeExpansion` because right now you can only increase the size of the volumes to expand them - you cannot decrease it.
+Strimzi will check that the storage class has this option set to `true` before it tries to resize any volumes.
+The [original blog post](https://strimzi.io/2019/02/28/resizing-persistent-volumes.html) explained in detail how does the resizing work:
+
+1. The size increase is requested by changing the `spec.resources.requests.storage` of the Persistent Volume Claim (PVC)
+2. Kubernetes will request resizing of the volumes from your infrastructure
+3. Once the resizing of the volume is finished, the pod using this volume needs to be restarted to allow the expansion of the file system
+
+All what Strimzi does is that it simplifies this process and takes care of it for you.
+As a user all you need to do is edit your Kafka custom resource and increase the requested storage size.
+You can increase the size of both Zookeeper and Kafka volumes.
+For example you can change the resource from:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+    storage:
+      type: persistent-claim
+      size: 100Gi
+      deleteClaim: false
+  zookeeper:
+    # ...
+    storage:
+      type: persistent-claim
+      size: 100Gi
+      deleteClaim: false
+```
+
+to:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+    storage:
+      type: persistent-claim
+      size: 200Gi
+      deleteClaim: false
+  zookeeper:
+    # ...
+    storage:
+      type: persistent-claim
+      size: 1000Gi
+      deleteClaim: false
+```
+
+And the rest will be taken care of by Strimzi.
+The cluster operator will automatically change the requested volume size in the PVCs and wait until the restart of the pod is required.
+Once the condition of the PVC is set to `FileSystemResizePending`, Strimzi will automatically restart the pod using this PVC.
```

Comment:
maybe adding the command line you used for getting the following status information?

Reply:
The users doesn't need to check the status of the PVCs. The Operator does it. Adding the commands might make it look more complicated than it is. I added a link to the older blog post which has more details about the resizing process.